src/os/ui/data/layercheckbox.js

```diff
@@ -0,0 +1,149 @@
+goog.module('os.ui.data.LayerCheckboxUI');
+
+const ConfirmUI = goog.require('os.ui.window.ConfirmUI');
+const DescriptorNode = goog.require('os.ui.data.DescriptorNode');
+const LayerSyncDescriptor = goog.require('os.data.LayerSyncDescriptor');
+const Module = goog.require('os.ui.Module');
+const TriState = goog.require('os.structs.TriState');
+const TriStateCheckboxCtrl = goog.require('os.ui.TriStateCheckboxCtrl');
+const triStateCheckboxDirective = goog.require('os.ui.triStateCheckboxDirective');
+
+const TriStateTreeNode = goog.requireType('os.structs.TriStateTreeNode');
+
+
+/**
+ * Tristate checkbox extension for enabling layers.
+ *
+ * @return {angular.Directive}
+ */
+const directive = () => {
+  const dir = triStateCheckboxDirective();
+  dir.controller = Controller;
+  return dir;
+};
+
+
+/**
+ * The element tag for the directive.
+ * @type {string}
+ */
+const directiveTag = 'layercheckbox';
+
+
+/**
+ * Add the directive to the module
+ */
+Module.directive(directiveTag, [directive]);
+
+
+/**
+ * Controller for the layer checkbox.
+ * @unrestricted
+ */
+class Controller extends TriStateCheckboxCtrl {
+  /**
+   * Constructor.
+   * @param {!angular.Scope} $scope The Angular scope.
+   * @param {!angular.JQLite} $element The root DOM element.
+   * @ngInject
+   */
+  constructor($scope, $element) {
+    super($scope, $element);
+
+    /**
+     * The minimum count of layers before prompting the user.
+     * @type {number}
+     */
+    this.minCount = $scope['minCount'] || 10;
+  }
+
+  /**
+   * @inheritDoc
+   */
+  toggle(e) {
+    if (this.scope) {
+      const item = /** @type {TriStateTreeNode} */ (this.scope['item']);
+
+      if (!this.getDisabled(item)) {
+        const children = item.getChildren();
+        const state = item.getState();
+
+        if (state == TriState.OFF) {
+          let count = 0;
+          const countLayerChildren = (child) => {
+            if (child instanceof DescriptorNode) {
+              const descriptor = child.getDescriptor();
+
+              if (descriptor instanceof LayerSyncDescriptor) {
+                const layers = descriptor.getOptions();
+
+                if (Array.isArray(layers)) {
```

Comment:
Is `Array.isArray` needed? `getOptions` should return an array or null, and on null we shouldn't adjust the count.

Reply:
Woops, was originally calling `getLayerOptions()` which is protected, but returns `{Array<Object>|Object}`. Didn't realize the return profile was different here.